### DISCUSSION TITLE
conditional_after in WeibullAFTFitter.predict_expectation()

### DISCUSSION THREAD

**Discussion by sve007** [CamDavidsonPilon/lifelines#1492] [Q&A] [answered]
Hi all,

Can someone enlighten me why e.g. the WeibullAFTFitter().predict_expectation() function does not have the parameter conditional_after? When using predict_median for these models I do have this option. Is there a good mathematical explanation or is it something that is not (yet) implemented?

Thanks

**Comment by CamDavidsonPilon** [comment, marked as answer]
It's not implemented, but the gist of it is that it would rely on an approximation algorithm. To get around this, try: 
```python
from lifelines.fitters import ParametricRegressionFitter
super(ParametricRegressionFitter, waft).predict_expecation(..., conditional_after=...)
```

Hacky, but I can make this better in the next release.

**Reply by sve007** [reply]
Thanks, I will give this a try. Good to know that there is not a mathematical reason that this wouldn't be possible.
